app/models/review.rb

```diff
@@ -0,0 +1,25 @@
+class Review < ApplicationRecord 
+   belongs_to :book
+   belongs_to :user
+
+  def self.average_rating(id)
+    where(book_id: id).average(:rating)
+  end
+
+  def self.highest_rating(id)
+    where(book_id: id).maximum(:rating)
+  end
+
+  def self.lowest_rating(id)
+    where(book_id: id).minimum(:rating)
+  end
+
+  def self.highest_rating_info(id)
+    where(book_id: id).order("rating DESC").first
+  end
+
+  def self.lowest_rating_info(id)
+    where(book_id: id).order("rating ASC").first
+  end
```

Comment:
I'd want all these analysis methods to be built on Book since they each apply to a specific instance of a book.  Any time I start a method by scoping to a specific instance of another object it tells me my SRP is off.